scipy/special/xsf/cephes/gamma.h

```diff
@@ -360,16 +375,22 @@ namespace cephes {
         }
         if (x > 0) {
             return 1.0;
-        } else {
-            fx = std::floor(x);
-            if (x - fx == 0.0) {
-                return 0.0;
-            } else if (static_cast<int>(fx) % 2) {
-                return -1.0;
-            } else {
-                return 1.0;
-            }
-        }
+	}
+	if (x == 0) {
+	    return std::copysign(1.0, x);
+	}
+	if (std::isinf(x)) {
+	    // x > 0 case handled, so x must be negative infinity.
+	    return std::numeric_limits<double>::quiet_NaN();
+	}
+	fx = std::floor(x);
+	if (x - fx == 0.0) {
+	    return std::numeric_limits<double>::quiet_NaN();
+	}
+	if (static_cast<int>(fx) % 2) {
+	    return -1.0;
+	}
```

Comment:
I'd add a comment à la: "sign of the gamma function for x in (-N,-N+1) is (-1)^N"